Document/0x05b-Basic-Security_Testing.md

```diff
@@ -207,6 +204,217 @@ Unlike static analysis, dynamic analysis is performed while executing the mobile
+
+> For both solutions you need to activate "Support invisible proxying" in Burp, in Proxy Tab/Options/Edit Interface.
+
+**iptables**
+
+You can use iptables on the Android device to  redirect all traffic to your interception proxy. The following command would redirect port 80 to your proxy running on port 8080
+
+```bash
+$ iptables -t nat -A OUTPUT -p tcp --dport 80 -j DNAT --to-destination <Your-Proxy-IP>:8080
+```
+
+Verify the iptables settings and check the IP and port.
+
+```bash
+$ iptables -t nat -L
+Chain PREROUTING (policy ACCEPT)
+target     prot opt source               destination
+
+Chain INPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain OUTPUT (policy ACCEPT)
+target     prot opt source               destination
+DNAT       tcp  --  anywhere             anywhere             tcp dpt:5288 to:<Your-Proxy-IP>:8080
+
+Chain POSTROUTING (policy ACCEPT)
+target     prot opt source               destination
+
+Chain natctrl_nat_POSTROUTING (0 references)
+target     prot opt source               destination
+
+Chain oem_nat_pre (0 references)
+target     prot opt source               destination
+```
+
+In case you want to reset the iptables configuration you can flush the rules:
+
+```bash
+$ iptables -t nat -F
+```
+
+**Ettercap**
+
+Read the chapter "Testing Network Communication" and the test case "Simulating a Man-in-the-Middle Attack" for further preparation and instructions for running ettercap.
+
+Your testing machine and the Android device must be connected to the same wireless network. Start ettercap with the following command, replacing the IP addresses below with the IP addresses of your Android device and the wireless network's gateway.
+
+```bash
+$ sudo ettercap -T -i en0 -M arp:remote /192.168.0.1// /192.168.0.105//
+```
+
+#### Network Monitoring/Sniffing
+
+ [Remotely sniffing all Android traffic in real-time is possible with tcpdump, netcat (nc), and Wireshark](https://blog.dornea.nu/2015/02/20/android-remote-sniffing-using-tcpdump-nc-and-wireshark/ "Android remote sniffing using Tcpdump, nc and Wireshark"). First, make sure that you have the latest version of [Android tcpdump](https://www.androidtcpdump.com/) on your phone. Here are the [installation steps](https://wladimir-tm4pda.github.io/porting/tcpdump.html "Installing tcpdump"):
+
+```
+# adb root
+# adb remount
+# adb push /wherever/you/put/tcpdump /system/xbin/tcpdump
+```
+
+If execution of `adb root` returns the  error `adbd cannot run as root in production builds`, install tcpdump as follows:
+
+```
+# adb push /wherever/you/put/tcpdump /data/local/tmp/tcpdump
+# adb shell
+# su
+$ mount -o rw,remount /system;
+$ cp /data/local/tmp/tcpdump /system/xbin/
+$ cd /system/xbin
+$ chmod 755 tcpdump
+```
```

Comment:
See previous, now you missed sh at the opening ```shell

Reply:
done